cmd/validator/validate_accounting.go

```diff
@@ -235,7 +236,7 @@ func callProcessor(processor Processor, addrInput string, config Params, results
 
 		result, err := processor.ProcessAddress(algodData, indexerData)
 
-		if err == nil && (result.Equal || i >= config.retries) {
+		if err == nil && (result.Equal || result.SameRound || i >= config.retries) {
```

Comment:
I think `result.Equal` is unnecessary.

Reply:
Since the reward base only changes every 30 rounds or so, most of the time it doesn't matter if we're on the same round. In that case `result.Equal` can be true and `result.SameRound` false.

Reply:
I would like to stay on the safe side and only return results with the same round. Efficiency shouldn't suffer since most of the time algod and indexer are synchronized.

Reply:
There isn't any risk because the comparison has to succeed for the test to pass. Why bother waiting and trying again if the test has already passed?

Reply:
Because two results having same data for different rounds doesn't mean they will have same data for the same round.

Reply:
If we were off by an entire round, then we would see failures when accounts are checked and indexer/algod are on the same round.

I wouldn't mind a strict mode flag which does this, but considering the test already takes something like 12 hours to run, I think we should let it run as fast as possible rather than worry about this case.

Reply:
Like I said before, I only saw dots in the output and no numbers, which means there are actually very few retries in practice. So, efficiency shouldn't suffer. But I would like to have more reliability.

Reply:
IIRC after the 12 hours, there will be hundreds or low thousands of retries.

Let's save this suggestion for a follow-up when we can also measure the impact

Reply:
I was lamenting the lack of retrys in all the old logs on my test machine, but I found some in nohup.out:
```
Number of errors: [112 / 9969050]
Retry count: 2365
Checks per second: 160.449854
Test duration: 17:15:46
```

Reply:
Ok, 2365 * 50ms = 2 minutes of additional time. I think, that's reasonable.

Reply:
By my calculation is was something between 15-39 minutes. The retry delay defaults to 1 second.

Reply:
You are right. But can we make the retry delay 100ms? We can also make the number of retries very large because now we will not retry that many times if there is a discrepancy.

Reply:
That doesn't account for the fact that every time a goroutines is waiting we aren't checking `160/<num-threads>` other accounts.

Also, only dots in the output doesn't mean that algod and indexer are synchronized. There's no indicator for matches that ignore that the round is mismatched.

Reply:
I thought that Indexer normally lags behind algod by several seconds. We would effectively be pausing the checks by that delta for every round. Hypothetically, lets say Indexer is lagging behind algod by 100ms and we're able to hit that time perfectly.

That would put our idle time at 100ms / 4.4s = 2.3%
In the run from above it took 17.25 hours. (17.25 * 1.023) = 17.25, so an extra ~24 minutes of downtime.

I think those numbers are probably pretty generous and the impact could be much larger.

Reply:
Got 79000 retries in my validation, so I guess it's pretty considerable. Can we still decrease retry timeout to 100ms and increase number of retries?

Reply:
Yeah, you can set those to whatever you want - there are CLI arguments